modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/policy/DefaultPolicyTemplateFactory.java

```diff
@@ -73,13 +77,38 @@ public PolicyTemplate createArtifact(Application application, PolicyTemplateDesc
     application.getRegionClassLoader().addClassLoader(policyClassLoader, NULL_CLASSLOADER_FILTER);
 
     List<ArtifactPlugin> artifactPlugins = createArtifactPluginList(policyClassLoader, resolvedPolicyPluginsDescriptors);
+    return new DefaultPolicyTemplate(policyClassLoader.getArtifactId(), descriptor, policyClassLoader,
+                                     artifactPlugins,
+                                     resolveOwnArtifactPlugins(application, descriptor, policyClassLoader, artifactPlugins));
+  }
 
-    validateArtifactLicense(policyClassLoader.getClassLoader(), artifactPlugins, licenseValidator);
+  // Need all the plugins that the policy itself depends on, while keeping a relationship with the appropriate classloader.
+  private List<ArtifactPlugin> resolveOwnArtifactPlugins(Application application, PolicyTemplateDescriptor descriptor,
```

Comment:
I have tried to understand why this needs to be done only in the context of Policies and not with Applications. Still not clear to me the reason but seems that descriptors have different plugin registered. As this fixes the issue we should merge this and I would add a comment to https://www.mulesoft.org/jira/browse/MULE-15842 which also related to hardcoded code for HTTP on policies.